### ISSUE TITLE
If you right-click and drag in the 3D editor window, then release, the editor will also crash.

### ISSUE THREAD

**Issue by OlexiyKravchuk** [godotengine/godot#33901]
**Godot version:**
Godot_v3.2-beta2_win64

**OS/device including version:**
W7 x64

**Steps to reproduce:**
If you right-click and drag in the 3D editor window, then release, the editor will also crash.

I spent half a day figuring out what was going on. it turned out that this happens in a very special case, with a certain combination of nodes and their hierarchy of instances of scenes.
I made a minimal demo project and a short video.

https://github.com/godotengine/godot/files/3899131/test.zip

https://youtu.be/_zptpFgQuC0

By the way, I want to note that the combination of nodes that I provided leads to such a result if, for example, replace at least one of these nodes, for example, Position 3D in a regular spatial one, or a Button for control, or a mesh instance for a spatial one, then a failure will not occur.

**Comment by KoBeWi** [comment]
Any more details? I can't reproduce it.

**Comment by aaronfranke** [comment]
Can't reproduce either, at least with the given instructions.

By "drag", do you just mean moving the mouse? Right-clicking in the 3D editor rotates the camera, so there is nothing visual on the screen that is dragging.

**Comment by OlexiyKravchuk** [comment]
In the space of the 3D viewport, I press the left mouse button and start dragging, an extrusion frame appears, and when I release the button, then GODOT crashes.

**Comment by OlexiyKravchuk** [comment]
I spent half a day figuring out what was going on. it turned out that this happens in a very special case, with a certain combination of nodes and their hierarchy of instances of scenes.
I made a minimal demo project and a short video.

[test.zip](https://github.com/godotengine/godot/files/3899131/test.zip)

https://youtu.be/_zptpFgQuC0

**Comment by KoBeWi** [comment]
Ok, I know what possibly happens. You try to select a 3D node that is a child of 2D node under 3D node and the editor just freaks out.

Here's the backtrace:
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] Dictionary::has (x:\inne\godot\source\godot\core\dictionary.cpp:135)
[1] Object::has_meta (x:\inne\godot\source\godot\core\object.cpp:1032)
[2] SpatialEditorViewport::_is_node_locked (x:\inne\godot\source\godot\editor\plugins\spatial_editor_plugin.cpp:824)
[3] SpatialEditorViewport::_select_region (x:\inne\godot\source\godot\editor\plugins\spatial_editor_plugin.cpp:524)
[4] SpatialEditorViewport::_sinput (x:\inne\godot\source\godot\editor\plugins\spatial_editor_plugin.cpp:1162)
[5] MethodBind1<SpatialEditorViewport,Ref<InputEvent> const &>::call (x:\inne\godot\source\godot\core\method_bind.gen.inc:867)
[6] Object::call (x:\inne\godot\source\godot\core\object.cpp:921)
[7] Object::emit_signal (x:\inne\godot\source\godot\core\object.cpp:1218)
[8] Object::emit_signal (x:\inne\godot\source\godot\core\object.cpp:1275)
[9] Viewport::_gui_call_input (x:\inne\godot\source\godot\scene\main\viewport.cpp:1626)
[10] Viewport::_gui_input_event (x:\inne\godot\source\godot\scene\main\viewport.cpp:2014)
[11] Viewport::input (x:\inne\godot\source\godot\scene\main\viewport.cpp:2792)
[12] Viewport::_vp_input (x:\inne\godot\source\godot\scene\main\viewport.cpp:1411)
[13] MethodBind1<Viewport,Ref<InputEvent> const &>::call (x:\inne\godot\source\godot\core\method_bind.gen.inc:867)
[14] Object::call (x:\inne\godot\source\godot\core\object.cpp:921)
[15] Object::call (x:\inne\godot\source\godot\core\object.cpp:847)
[16] SceneTree::call_group_flags (x:\inne\godot\source\godot\scene\main\scene_tree.cpp:275)
[17] SceneTree::input_event (x:\inne\godot\source\godot\scene\main\scene_tree.cpp:430)
[18] InputDefault::_parse_input_event_impl (x:\inne\godot\source\godot\main\input_default.cpp:442)
[19] InputDefault::parse_input_event (x:\inne\godot\source\godot\main\input_default.cpp:260)
[20] InputDefault::flush_accumulated_events (x:\inne\godot\source\godot\main\input_default.cpp:679)
[21] OS_Windows::process_events (x:\inne\godot\source\godot\platform\windows\os_windows.cpp:2447)
[22] OS_Windows::run (x:\inne\godot\source\godot\platform\windows\os_windows.cpp:3142)
[23] widechar_main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:162)
[24] _main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:184)
[25] main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:196)
[26] __scrt_common_main_seh (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283)
[27] BaseThreadInitThunk
-- END OF BACKTRACE --
```

**Comment by OlexiyKravchuk** [comment]
By the way, I want to note that the combination of nodes that I provided leads to such a result if, for example, replace at least one of these nodes, for example, Position 3D in a regular spatial one, or a Button for control, or a mesh instance for a spatial one, then a failure will not occur.

**Comment by aaronfranke** [comment]
It's an easily fixable regression from #30602.

The fix is in #33964, but one note is that right clicking has nothing to do with it, so the title of this issue is a bit inaccurate. It fixes the issue in KoBeWi's stacktrace, reproducible with left clicking.
